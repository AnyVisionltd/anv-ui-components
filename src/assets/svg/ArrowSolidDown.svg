<svg xmlns="http://www.w3.org/2000/svg" id="prefix__Arrow_solid_down" width="24" height="24" data-name="Arrow solid down" viewBox="0 0 24 24">
    <path id="prefix__Shape" d="M0 0H24V24H0z" style="fill:#02234b;opacity:0"/>
    <path id="prefix__Arrow_solid_down-2" d="M5.247.86a1 1 0 0 1 1.505 0l3.8 4.339A1 1 0 0 1 9.8 6.857H2.2A1 1 0 0 1 1.451 5.2z" data-name="Arrow solid down" transform="rotate(180 9 8.116)"/>
</svg>
