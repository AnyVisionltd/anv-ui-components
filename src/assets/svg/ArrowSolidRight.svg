<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
    <path d="M5.247.86a1 1 0 0 1 1.505 0l3.8 4.339A1 1 0 0 1 9.8 6.857H2.2A1 1 0 0 1 1.451 5.2z" transform="rotate(90 5.088 11.09)"/>
</svg>
