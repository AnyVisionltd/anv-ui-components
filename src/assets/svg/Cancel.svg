<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16">
    <g id="prefix__Cancel" transform="translate(-82 -8)">
        <path id="prefix__Shape" d="M0 0H16V16H0z" transform="translate(82 8)" style="opacity:0"/>
        <path id="prefix__Path_2223" d="M12.13 11l2.634-2.634a.8.8 0 1 0-1.131-1.131L11 9.868 8.365 7.234a.8.8 0 1 0-1.131 1.131L9.868 11l-2.634 2.633a.8.8 0 1 0 1.131 1.131L11 12.13l2.634 2.634a.8.8 0 1 0 1.131-1.131z" data-name="Path 2223" transform="translate(79.001 5.001)"/>
    </g>
</svg>
