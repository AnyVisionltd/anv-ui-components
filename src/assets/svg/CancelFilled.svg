<svg xmlns="http://www.w3.org/2000/svg" id="prefix__Cancel_X" width="24" height="24" data-name="Cancel X" viewBox="0 0 24 24">
    <path id="prefix__Shape" d="M0 0H24V24H0z" style="opacity:0;fill:#02234b"/>
    <path id="prefix__Subtraction_5" d="M8 16A8 8 0 0 1 2.344 2.343a8 8 0 1 1 11.312 11.31A7.941 7.941 0 0 1 8 16zm0-6.871l2.633 2.634a.841.841 0 0 0 .316.2.806.806 0 0 0 .251.037.791.791 0 0 0 .564-.234A.8.8 0 0 0 12 11.2a.793.793 0 0 0-.234-.564L9.131 8l2.633-2.634a.8.8 0 0 0 0-1.131.8.8 0 0 0-1.131 0L8 6.87 5.365 4.236a.8.8 0 0 0-1.131 0 .8.8 0 0 0 0 1.131L6.869 8l-2.635 2.635A.792.792 0 0 0 4 11.2a.8.8 0 0 0 .8.8.791.791 0 0 0 .564-.234L8 9.131z" data-name="Subtraction 5" transform="translate(4 4)"/>
</svg>
