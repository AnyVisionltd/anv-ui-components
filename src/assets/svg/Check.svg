<svg xmlns="http://www.w3.org/2000/svg" id="prefix__check" width="24" height="24" viewBox="0 0 24 24">
    <path id="prefix__Shape" d="M0 0H24V24H0z" style="opacity:0;fill:#00d0a0"/>
    <path id="prefix__check-2" d="M18.169 7.2a.961.961 0 0 0-1.364 0l-7.157 7.164-3.006-3.016a.982.982 0 1 0-1.364 1.412l3.689 3.689a.961.961 0 0 0 1.364 0l7.838-7.839a.961.961 0 0 0 0-1.412z" data-name="check" transform="translate(.272 -.163)" />
</svg>
