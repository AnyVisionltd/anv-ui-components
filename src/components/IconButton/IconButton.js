import React from 'react'
import classNames from 'classnames'
import propTypes from 'prop-types'
import colors from '@anyvision/style-guide/abstracts/_colors.scss'
import { Button } from '../Button'
import styles from './IconButton.module.scss'

const styleGuideColors = Object.keys(colors)

const IconButton = ({ icon: SvgIcon, ...buttonProps }) => {
  const classes = classNames(
    styles.iconButton,
  )

  return (
    <Button className={ classes } { ...buttonProps }>
      <SvgIcon className={ styles.svgIcon } />
    </Button>
  )
}

IconButton.defaultProps = {
  color: 'primary',
  size: 'large',
  variant: 'fill',
  disabled: false,
  onClick: () => {},
  type: 'button',
}

IconButton.propTypes = {
  /** Svg icon */
  icon: propTypes.elementType.isRequired,
  /** The color of the button. */
  color: propTypes.oneOf(styleGuideColors),
  /** The size of the button. */
  size: propTypes.oneOf(['small', 'large']),
  /** The variant of the button. */
  variant: propTypes.oneOf(['fill', 'outline', 'ghost']),
  /** If true, the button will be disabled. */
  disabled: propTypes.bool,
  /** Callback when click. */
  onClick: propTypes.func,
  /** Defines HTML button type attribute. */
  type: propTypes.oneOf(['button', 'reset', 'submit']),
}

export default IconButton
