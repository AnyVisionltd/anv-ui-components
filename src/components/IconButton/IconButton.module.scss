@import "~@anyvision/style-guide";

.iconButton {
  height: av-size(sz-32);
  width: av-size(sz-32);
  padding: 0;
}

.svgIcon {
  fill: av-color(purewhite);
}

