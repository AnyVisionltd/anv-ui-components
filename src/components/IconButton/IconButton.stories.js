import React from 'react'
import { select, boolean } from '@storybook/addon-knobs'
import { ReactComponent as SunIcon } from '../../assets/svg/Sun.svg'
import IconButton from './IconButton'
import styles from '../../styles/storybook/index.module.scss'
import { centerDecorator } from '../../utils/storybook/decorators'

export default {
  title: 'IconButton',
  component: IconButton,
  decorators: [centerDecorator],
}

export const Default = () => (
  <IconButton>
    <SunIcon />
  </IconButton>
)

export const variants = () => (['fill', 'outline', 'ghost'].map((variant) => (
  <IconButton className={ styles.microMargin } key={ variant } variant={ variant }>
    <SunIcon />
  </IconButton>
))
)

export const sizes = () => (
  <>
    <IconButton className={ styles.microMargin }><SunIcon /></IconButton>
    <IconButton className={ styles.microMargin } size="large"><SunIcon /></IconButton>
  </>
)


export const disable = () => (
  <>
    <IconButton className={ styles.microMargin }><SunIcon /></IconButton>
    <IconButton className={ styles.microMargin } disabled><SunIcon /></IconButton>
  </>
)

export const withClassName = () => (
  <div className={ styles.marginFlexContainer }>
    { /* background-color: av-color(success); */ }
    <IconButton className={ styles.successBackgroundColor }>
      <SunIcon />
    </IconButton>
    { /* color: av-color(alert); */ }
    <IconButton className={ styles.alertColor } variant="outline">
      <SunIcon />
    </IconButton>
    { /* color: av-color(error); */ }
    <IconButton className={ styles.errorColor } variant="ghost">
      <SunIcon />
    </IconButton>
  </div>
)

export const playGround = () => (
  <IconButton
    size={ select('size', ['small', 'large'], 'small') }
    variant={ select('variant', ['fill', 'outline', 'ghost'], 'fill') }
    disabled={ boolean('disabled', false) }
  >
    <SunIcon />
  </IconButton>
)
