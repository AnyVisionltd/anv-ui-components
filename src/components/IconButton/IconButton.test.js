import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { ReactComponent as SvgIcon } from '../../../jest/assets/svgIcon.svg'
import IconButton from './IconButton'

describe('<IconButton />', () => {
	it('should render children into button', () => {
		const { container } = render(<IconButton><SvgIcon /></IconButton>)
		const node = container.querySelector('svg')
		expect(node).toBeTruthy()
	})

	describe('disabled', () => {
		it('should NOT be disabled by default', () => {
			const { getByRole } = render(<IconButton><SvgIcon /></IconButton>)
			const node = getByRole('button')
			expect(node.disabled).toBe(false)
		})

		it('should be disabled when disabled pass', () => {
			const { getByRole } = render(<IconButton disabled><SvgIcon /></IconButton>)
			const node = getByRole('button')
			expect(node).toBeDisabled()
		})
	})

	it('should call onClick', () => {
		const handleClick = jest.fn()
		const { getByRole } = render(
			<IconButton onClick={ handleClick }><SvgIcon /></IconButton>,
		)
		const node = getByRole('button')
		fireEvent.click(node)
		expect(handleClick).toBeCalled()
	})
})
