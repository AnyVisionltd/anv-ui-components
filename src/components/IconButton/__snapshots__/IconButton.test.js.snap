// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Button Colors 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <button
      className="button large primary-fill microMargin"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      primary
    </button>
  </div>
</div>
`;

exports[`Storyshots Button Default 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <button
      className="button large primary-fill"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Text
    </button>
  </div>
</div>
`;

exports[`Storyshots Button Disable 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <button
      className="button large primary-fill microMargin"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Enable
    </button>
    <button
      className="button large primary-fill microMargin"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Disabled
    </button>
  </div>
</div>
`;

exports[`Storyshots Button Play Ground 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <button
      className="button large primary-fill"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Button Text
    </button>
  </div>
</div>
`;

exports[`Storyshots Button Sizes 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <button
      className="button large primary-fill microMargin"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Large
    </button>
    <button
      className="button small primary-fill microMargin"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      small
    </button>
  </div>
</div>
`;

exports[`Storyshots Button Variants 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <button
      className="button large primary-fill microMargin"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      fill
    </button>
    <button
      className="button large primary-outline microMargin"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      outline
    </button>
    <button
      className="button large primary-ghost microMargin"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      ghost
    </button>
  </div>
</div>
`;

exports[`Storyshots Button With Icon 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <button
      className="button large primary-fill microMargin"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        className="startIcon"
      />
      Start
    </button>
  </div>
</div>
`;

exports[`Storyshots Checkbox Colors 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <label
      className="checkbox primary checkbox-primary checked microMargin"
    >
      <input
        checked={true}
        className="parent"
        disabled={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="iconContainer"
      >
        <svg />
      </span>
    </label>
  </div>
</div>
`;

exports[`Storyshots Checkbox Default 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <label
      className="checkbox primary checkbox-primary"
      htmlFor="checkbox-playground"
    >
      <input
        checked={false}
        className="parent"
        disabled={false}
        id="checkbox-playground"
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="iconContainer"
      />
    </label>
    <label
      className="checkboxLabel"
      htmlFor="checkbox-playground"
    >
      Check me out!
    </label>
  </div>
</div>
`;

exports[`Storyshots Checkbox Playground 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <label
      className="checkbox primary checkbox-primary"
      htmlFor="checkbox-playground"
    >
      <input
        checked={false}
        className="parent"
        disabled={false}
        id="checkbox-playground"
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="iconContainer"
      />
    </label>
    <label
      className="checkboxLabel"
      htmlFor="checkbox-playground"
    >
      Hit me!
    </label>
  </div>
</div>
`;

exports[`Storyshots Checkbox States 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <label
      className="checkbox primary checkbox-primary checked microMargin"
    >
      <input
        checked={true}
        className="parent"
        disabled={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="iconContainer"
      >
        <svg />
      </span>
    </label>
    <label
      className="checkbox primary checkbox-primary microMargin"
    >
      <input
        checked={false}
        className="parent"
        disabled={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="iconContainer"
      />
    </label>
    <label
      className="checkbox primary checkbox-primary indeterminate microMargin"
    >
      <input
        checked={false}
        className="parent"
        disabled={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="iconContainer"
      >
        <svg />
      </span>
    </label>
    <label
      className="checkbox primary checkbox-primary isDisabled microMargin"
    >
      <input
        checked={false}
        className="parent"
        disabled={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="iconContainer"
      />
    </label>
  </div>
</div>
`;

exports[`Storyshots IconButton Colors 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <button
      className="button large primary-fill iconButton small fill microMargin"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg />
    </button>
  </div>
</div>
`;

exports[`Storyshots IconButton Default 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <button
      className="button large primary-fill iconButton small fill"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg />
    </button>
  </div>
</div>
`;

exports[`Storyshots IconButton Disable 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <button
      className="button large primary-fill iconButton small fill microMargin"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg />
    </button>
    <button
      className="button large primary-fill iconButton small fill microMargin"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      <svg />
    </button>
  </div>
</div>
`;

exports[`Storyshots IconButton Play Ground 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <button
      className="button large primary-fill iconButton small fill"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg />
    </button>
  </div>
</div>
`;

exports[`Storyshots IconButton Sizes 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <button
      className="button large primary-fill iconButton small fill microMargin"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg />
    </button>
    <button
      className="button large primary-fill iconButton large fill microMargin"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg />
    </button>
  </div>
</div>
`;

exports[`Storyshots IconButton Variants 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <button
      className="button large primary-fill iconButton small fill microMargin"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg />
    </button>
    <button
      className="button large primary-outline iconButton small outline microMargin"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg />
    </button>
    <button
      className="button large primary-ghost iconButton small ghost microMargin"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg />
    </button>
  </div>
</div>
`;
